### PR TITLE
[BUDI-8433] Default value column UI

### PR DESCRIPTION
**Pull request by samwho**
## Description

This PR creates the UI for default values. It's currently behind a feature flag so it will be hidden in production but shown in QA.

https://github.com/user-attachments/assets/91e29968-2938-4054-9fd5-a1082389ec26

### CODE REVIEW

File: packages/builder/src/components/backend/DataTable/modals/CreateEditColumn.svelte [hunk 1]
```diff
@@ -165,11 +168,20 @@
     editableColumn?.type !== AUTO_TYPE &&
     editableColumn?.type !== JSON_TYPE &&
     !editableColumn.autocolumn
+  $: canHaveDefault =
+    isEnabled(TENANT_FEATURE_FLAGS.DEFAULT_VALUES) &&
+    (editableColumn?.type === FieldType.NUMBER ||
```
Comment (adrinr) [inline packages/builder/src/components/backend/DataTable/modals/CreateEditColumn.svelte:173]: Is this controlled only via frontend? Don't we have a backend validation?

Comment (melohagan) [reply packages/builder/src/components/backend/DataTable/modals/CreateEditColumn.svelte:173]: NAB: could also refactor into `[FieldType.NUMBER, FieldType...].includes(editableColumn?.type)`

**Review by melohagan — APPROVED** [review]
LGTM - awesome feature! 😄 

I noticed that the default value supports bindings. Have you tried a few scenarios?

I wonder if this would allow you to do auto-increments, or other sequencing formulas?

File: packages/builder/src/components/backend/DataTable/modals/CreateEditColumn.svelte (hunk 1 above)
Comment (samwho) [reply packages/builder/src/components/backend/DataTable/modals/CreateEditColumn.svelte:173]: Currently only controlled via the frontend. Should be easy to add backend validation if you thought it was worth it, but at the time I felt like it was easier to reason about this only being toggled in the frontend.

Comment (samwho) [reply packages/builder/src/components/backend/DataTable/modals/CreateEditColumn.svelte:173]: @melohagan In an ideal would I'd like to do a broader refactor of this file to rely far less on checking hardcoded field types, and instead have the FieldSchema dictate what can and cannot be done with a field, then genericise this component using the FieldSchema data.

**Comment by samwho** [comment]
@melohagan there are some simple examples of using bindings as default values in `row.spec.ts` that I added in https://github.com/Budibase/budibase/pull/14177/files#diff-428e0cad26d239dd58cf676e4dfef825a1caca2ee998ef8183dfc44e9103572eR681. If you've got more complex ideas in mind it might be a neat idea to make testcases for them to make sure they work. 🙂

File: packages/builder/src/components/backend/DataTable/modals/CreateEditColumn.svelte (hunk 1 above)
Comment (adrinr) [reply packages/builder/src/components/backend/DataTable/modals/CreateEditColumn.svelte:173]: Just worried of having inconsistencies if these values are set somehow. It it's not too hard, having a helper in shared-core that is used both in front and back would be great

**Review by adrinr — APPROVED** [review]
LGTM, just one final comment

File: packages/builder/src/components/backend/DataTable/modals/CreateEditColumn.svelte (hunk 1 above)
Comment (samwho) [reply packages/builder/src/components/backend/DataTable/modals/CreateEditColumn.svelte]: Have had a go at what I think you're looking for, PTAL 🙂

**Pull request merged**